File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void build(int size);

int main(void)
{
    // Initialize the height variable
    int height;

    // Promt the user to enter the height between 1 and 8
    do
    {
        height = get_int("Height: ");
    }
    while ((height < 1) || (height > 8));

    // Call the funtion to build the pyramid of a certain height
    build(height);

}

// Builds the pyramid of a certain height
void build(int size)
{
    // For each row
    for (int i = 1; i < size + 1; i++)
    {
        // Calculate the number of spaces on each row
        int spaces = size - i;

        // Spaces for each column
        for (int j = 0; j < spaces; j++)
        {
            // Print spaces
            printf(" ");
        }

        // Hashes for the first pyramid
        for (int k = 0; k < i; k++)
        {
            // Print hashes
            printf("#");
        }

        // Print a gap
        printf("  ");

        // Hashes for the second pyramid
        for (int k = 0; k < i; k++)
        {
            // Print hashes
            printf("#");
        }

        // Go to the next line
        printf("\n");
    }

}

File: sentimental-mario-more/mario.py
# TODO
from cs50 import get_int


# Check if height is valid number between 1 and 8
def check_height(number):
    return number >= 1 and number <= 8


# Print a pyramid of a given height, a gap between two parts is 2 spaces
def build(height):
    for i in range(1, height + 1):
        n_spaces = height - i
        print((n_spaces * " ") + ("#" * i) + "  " + ("#" * i))


height = 0
# Keep prompting user untill he enters a valid height
while not check_height(height):
    height = get_int("Height: ")

# Print a pyramid of a given height
build(height)
